fix `powershell -version` and help

SteveL-MSFT opened the pull request:
`PowerShell -v` behavior updated to align with other tools like git, curl, and bash where args after `-v` are silently ignored.

Built-in help updated to reflect changes we've made to the console host in PSCore6 removing unsupported parameters.

Doc update aligned with these changes (and general cleanup of parameters we don't support in PSCore6) https://github.com/PowerShell/PowerShell-Docs/pull/1707
<!--

If you are a PowerShell Team member, please make sure you choose the Reviewer(s) and Assignee for your PR.
If you are not from the PowerShell Team, you can leave the fields blank and the Maintainers will choose them for you. If you are familiar with the team, feel free to mention some Reviewers yourself.

For more information about the roles of Reviewer and Assignee, refer to [CONTRIBUTING.md](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md).

-->

SteveL-MSFT committed:
[feature]
removed test that is no longer valid due to change in -version behavior to return error

daxian-dbw commented:
The same test didn't fail in AppVeyor. In AppVeyor, the failed test is `Version should write an error if a value is present`, see https://ci.appveyor.com/project/PowerShell/powershell/build/6.0.0-beta.7-5642#L384

On Travis, a different test failed: `Version should ignore other parameters`. See https://travis-ci.org/PowerShell/PowerShell/jobs/281514325#L1461

So probably we shouldn't just remove that test.

SteveL-MSFT commented:
We still need to remove this test as the change is to return an error if there's anything after `-version` whereas previously we silently ignored it.  I think the current behavior is correct.

SteveL-MSFT commented:
Need to fix the other test and also make a change to the `-v` handling.  I should have caught this in the code review.  Apologies.

SteveL-MSFT committed:
[feature]
fixed `-v X` so that it errors out correctly with proper exit code
updated test to catch this

iSazonov commented on test/powershell/Host/ConsoleHost.Tests.ps1:
```diff
@@ -171,20 +171,11 @@ Describe "ConsoleHost unit tests" -tags "Feature" {
-            $expectedError = (Get-Content $tempFile)[0]
-
-            $expectedError | Should Match $versionValue
+            $observed = & $powershell -version $versionValue 2>&1
```
😕 I skipped [Feature] in previous PR. I saw "CI" on top the test file but we have "Feature" tests below. 
 _We need an idea how exclude this in future._
(I wonder that the test was passed locally.)

In CI log I see messages:
```
[00:08:17] Usage of '-Version -command' is not supported. '-Version' currently only returns the current PowerShell version.
[00:08:17] +
[00:08:17] Usage of '-Version abrakadabra' is not supported. '-Version' currently only returns the current PowerShell version.
```
The first message looks incorrect. Maybe: "Error in '-Version {0}'. Now '-Version' shows the current PowerShell version and don't accept an argument."

Also previous test helped us to catch the problem - so I prefer to don't remove it because it may test another code path in future.

SteveL-MSFT replied:
The test I removed was when we supported:

```bash
poweshell -version -command 1+1
```

I think this syntax doesn't make sense as what it ended up doing was outputting the version string and ignoring everything after it.  It seems that this should have caught an unintentional user error which is why I removed that test since the new behavior prevents that old behavior.

I agree that the error message could be better particularly in the above scenario.  I'll update it.

SteveL-MSFT replied:
I think I'll change it to

```none
Unsupported value '{0}' passed to `-Version`. '-Version' currently only returns the current PowerShell version and must be used standalone.
```

iSazonov reviewed (commented):
Leave a comment

SteveL-MSFT committed:
[feature]
address PR feedback

iSazonov commented on src/Microsoft.PowerShell.ConsoleHost/resources/CommandLineParameterParserStrings.resx:
```diff
@@ -273,6 +273,6 @@ EXAMPLES
   </data>
   <data name="DeprecatedVersionParameter" xml:space="preserve">
-    <value>Usage of '-Version {0}' is not supported. '-Version' currently only returns the current PowerShell version.</value>
+    <value>Unsupported value '{0}' passed to `-Version`. '-Version' only returns the current PowerShell version and must be used standalone.</value>
```
I still unlike this:
1. `-Version`. '-Version'  - looks bad.
2. Unsupported value - implies that there are supported values.

SteveL-MSFT replied:
If we follow the example of some other tools like: git, curl, bash (although each of them use `--version`), they basically ignore anything after `--version` and simply output the version string.  Perhaps that's the right behavior.

SteveL-MSFT replied:
For now, I think I'll follow the other tools and will update the documentation as well.

iSazonov replied:
Initial idea for PowerShell was to inform users that we change -Version semantics. So I write information error and PowerShell version too.

SteveL-MSFT committed:
[feature]
make -v have behavior consistent with other tools like git, curl, bash where args after -v are ignored
removed duplicate but not exactly help text that wasn't being used in the resource file
removed parameters from help that are not support in PSCore6

SteveL-MSFT commented:
@iSazonov if you're fine with the changes, can you merge so we can unblock other PRs?  If you have specific concerns with the expected behavior, I think we can continue that discussion in a new issue.  Thanks!

iSazonov commented:
One concern is that we show full help in case of an error.
It's very annoying in interactive session if I typo - I'm forced to scroll back to see error message and lost my history before eyes. I'd prefer to see an error message and shot hint "Use powershell -? or -h to get more help" .

SteveL-MSFT commented:
@iSazonov you referring to existing behavior and not something I introduced, correct?

iSazonov commented:
Yes, I wanted to say that we need more review and cleanups in the file. I see:
1. #if !CORE
2. _ShowHelp=true
3. maybe formattings

SteveL-MSFT commented:
@iSazonov can submit a new PR?  the `#if CORECLR` I was going to take care of as part of https://github.com/PowerShell/PowerShell/issues/4953

iSazonov commented:
I'll try to review on next week.

iSazonov commented:
@SteveL-MSFT Please clarify. In #4834 we decided to write an error to inform users about changing -Version behavior. In the PR we remove the error. Should we write in #4834 that we reject the request now?

SteveL-MSFT commented:
@iSazonov added comment and closed https://github.com/PowerShell/PowerShell/issues/4834

1. we should be consistent and, in general, use existing norms like `bash`
2. we should not output help either
3. agree with the proposed changes
4. absolutely!
